mlflow/models/evaluation/default_evaluator.py

```diff
@@ -1877,6 +1877,18 @@ def evaluate(
         predictions=None,
         **kwargs,
     ):
+        if model is None and predictions is None and dataset.predictions_data is None:
+            raise MlflowException(
+                message=(
+                    "Either a model or set of predictions must be specified in order to use the"
+                    " default evaluator. Either specify the `model` parameter, the `predictions`"
+                    " parameter, an MLflow dataset containing the `predictions` column name"
+                    " (via the `data` parameter), or a different evaluator (via the `evaluators`"
+                    " parameter)."
+                ),
+                error_code=INVALID_PARAMETER_VALUE,
+            )
```

Comment:
See https://github.com/mlflow/mlflow/pull/11831/files#r1580245904. Moved the exception into the default evaluator.